### PR TITLE
DO NOT MERGE: RHOARDOC-1604 reactive procedure modules

### PR DESCRIPTION
**Pull request by inoxx03**
This PR contains the docs procedure modules for the Spring Reactive release (part of SB 2.1.6).
This branch is the base for PR #1300 (reactive Concept Modules).
Ensure that PR #1300 is merged before merging this PR.

### CODE REVIEW

File: docs/topics/proc_using-oauth2-with-reactive-http-web-service-webflux.adoc [hunk 1]
```diff
@@ -0,0 +1,108 @@
+[id='using-oauth2-authentication-with-a-reactive-spring-boot-webflux-application_{context}']
+= Using OAuth2 authentication in a reactive {SpringBoot} application with WebFlux
+
+Set up OAuth2 for your {SpringBoot} Webflux application and authenticate with it using your client ID and client secret.
```
Comment (huntley) [inline docs/topics/proc_using-oauth2-with-reactive-http-web-service-webflux.adoc]: Maybe reword like: Configure OAuth2 to authenticate for your {SpringBoot} Webflux application using your client ID and client secret.

File: docs/topics/proc_using-oauth2-with-reactive-http-web-service-webflux.adoc [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+* Java JDK 8 or Java JDK 11 installed
+* A GitHub account
+* Maven installed
+* A Maven-based application project link:{link-guide-spring-boot}#configuring-your-application-to-use-spring-boot_spring-boot[configured to use Spring Boot]
+
+
+.Procedure
+
+. link:https://github.com/settings/developers[Register a new OAuth 2 application] on your Github account.
+Ensure that you provide the following values in the registration form:
```
Comment (huntley) [inline docs/topics/proc_using-oauth2-with-reactive-http-web-service-webflux.adoc]: I suggest either making this a single sentence with the above: Register a new OAuth 2 application on your Github account and provide the following values in the registration form:

or drop the "Ensure that you"

File: docs/topics/proc_using-oauth2-with-reactive-http-web-service-webflux.adoc [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+* Homepage URL: `http://localhost:8080`
+* Authorization callback URL: `http://localhost:8080/login/oauth2/code/github`
++
+Ensure that you save the client ID and a client secret that you receive upon completing the registration.
```
Comment (huntley) [inline docs/topics/proc_using-oauth2-with-reactive-http-web-service-webflux.adoc]: I think you can do without the "Ensure that you"

File: docs/topics/proc_using-oauth2-with-reactive-http-web-service-webflux.adoc [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+. Add the following dependencies in the `pom.xml` file of your project:
+
+* `vertx-spring-boot-starter-http`
+* `spring-boot-starter-security`
+* `spring-boot-starter-oauth2-client`
+* `reactor-netty`
+
+Note that the `reactor-netty` client is required to ensure that `spring-boot-starter-oauth2-client` works properly.
```
Comment (huntley) [inline docs/topics/proc_using-oauth2-with-reactive-http-web-service-webflux.adoc]: You can remove "Note that".

File: docs/topics/proc_using-oauth2-with-reactive-http-web-service-webflux.adoc [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+  <dependencies>
+  ...
+include::resources/spring-boot/reactive-examples/example-0auth2-with-reactive-http-application-webflux-pom.xml[tags=http-service-oauth2-pom-dependencies]
+  ...
+  <dependencies>
+...
+</project>
+----
+
+. Create an endpoint controller class for your application:
```
Comment (huntley) [inline docs/topics/proc_using-oauth2-with-reactive-http-web-service-webflux.adoc]: This is labelled Step 1 in the rendered output. I think it should be step 3.

File: docs/topics/proc_using-oauth2-with-reactive-http-web-service-webflux.adoc [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+
+. Create the main class of your application:
++
+.`HttpSecuritySampleApplication.java`
+[source,java,options="nowrap",subs="attributes+"]
+----
+include::resources/spring-boot/reactive-examples/example-oauth2-webflux-application.java[]
+----
+
+. Create a YAML configuration file to store the OAuth2 client ID and client secret you received from GitHub upon registering your application.
```
Comment (huntley) [inline docs/topics/proc_using-oauth2-with-reactive-http-web-service-webflux.adoc]: Remove "upon registering your application."

File: docs/topics/proc_using-server-sent-events-spring-reactive-app.adoc [hunk 7]
```diff
@@ -0,0 +1,99 @@
+[id='using-server-sent-events-in-a-spring-reactive-application_{context}']
+= Using Server-Sent Events in a Spring Reactive application
+
+Create a simple service that request and receives a stream of server-sent events (SSE)
```
Comment (huntley) [inline docs/topics/proc_using-server-sent-events-spring-reactive-app.adoc]: Request should be plural. Period after (SSE).

File: docs/topics/proc_using-server-sent-events-spring-reactive-app.adoc [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+The service accepts a HTTP request and responds by returning a continuous stream of events until you cancel the request.
```
Comment (huntley) [inline docs/topics/proc_using-server-sent-events-spring-reactive-app.adoc]: an HTTP request

File: docs/topics/proc_using-server-sent-events-spring-reactive-app.adoc [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+  <dependencies>
+  ...
+include::resources/spring-boot/reactive-examples/example-sse-service-pom.xml[tags=sse-service-pom-dependencies]
+  ...
+  <dependencies>
+...
+</project>
+----
+
+. Create the main class of your application:
```
Comment (huntley) [inline docs/topics/proc_using-server-sent-events-spring-reactive-app.adoc]: Rendered as step 1, should be step 2.

File: docs/topics/proc_using-server-sent-events-spring-reactive-app.adoc [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+
+.`SseExampleApplication.java`
+[source,java,options="nowrap",subs="attributes+"]
+----
+include::resources/spring-boot/reactive-examples/example-sse-application.java[]
+----
+
+. Create a Server-sent Event controller class for your application.
```
Comment (huntley) [inline docs/topics/proc_using-server-sent-events-spring-reactive-app.adoc]: Rendered as step 1, should be step 3.

File: docs/topics/proc_using-server-sent-events-spring-reactive-app.adoc [hunk 11]
```diff
@@ -0,0 +1,99 @@
...
+In this example, the class generates a stream of random integers and outputs them into a terminal application.   
+
+.`SseController.java`
+[source,java,options="nowrap",subs="attributes+"]
+----
+include::resources/spring-boot/reactive-examples/example-sse-service-sse-controller.java[]
+----
+
+. OPTIONAL: Run and test your application locally
```
Comment (huntley) [inline docs/topics/proc_using-server-sent-events-spring-reactive-app.adoc]: Period at end of sentence.

File: docs/topics/proc_using-server-sent-events-spring-reactive-app.adoc [hunk 12]
```diff
@@ -0,0 +1,99 @@
...
+[source,java,options="nowrap",subs="attributes+"]
+----
+include::resources/spring-boot/reactive-examples/example-sse-service-sse-controller.java[]
+----
+
+. OPTIONAL: Run and test your application locally 
+
+[source,bash,options="nowrap",subs="attributes+"]
+----
+$ cd myApp
```
Comment (huntley) [inline docs/topics/proc_using-server-sent-events-spring-reactive-app.adoc]: Should be step A

File: docs/topics/proc_using-server-sent-events-spring-reactive-app.adoc [hunk 13]
```diff
@@ -0,0 +1,99 @@
...
+$ mvn clean package
+----
+
+.. Start your application from the command line:
++
+[source,bash,options="nowrap",subs="attributes+"]
+----
+$ java -jar target/vertx-spring-boot-sample-sse.jar 
+----
+.. In a new terminal window, issue a HTTP request to `localhost`.
```
Comment (huntley) [inline docs/topics/proc_using-server-sent-events-spring-reactive-app.adoc]: an HTTP request

**Comment by tradej** [comment]
@inoxx03, please push fixes into [tradej:reactive-resolution](https://github.com/tradej/appdev-documentation/tree/reactive-resolution).

**Comment by tradej** [comment]
Closing in favour of #1307.
